src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx

```diff
@@ -99,23 +99,36 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
   const highestBalanceAccountId = useAppSelector(state =>
     selectHighestBalanceAccountIdByStakingId(state, highestBalanceAccountIdFilter),
   )
-  const defaultAccountId = useAppSelector(state => selectFirstAccountIdByChainId(state, ethChainId))
-  const opportunityDataFilter = useMemo(
-    () => ({
+  const defaultAccountId = useAppSelector(state => selectFirstAccountIdByChainId(state, chainId))
+  const maybeAccountId = useMemo(
+    () => accountId ?? highestBalanceAccountId ?? defaultAccountId,
+    [accountId, defaultAccountId, highestBalanceAccountId],
+  )
+
+  useEffect(() => {
+    if (!maybeAccountId) return
+    handleAccountIdChange(maybeAccountId)
+  }, [handleAccountIdChange, maybeAccountId])
+
+  const opportunityDataFilter = useMemo(() => {
+    if (!maybeAccountId?.length) return
+
+    return {
       userStakingId: serializeUserStakingId(
-        (accountId ?? highestBalanceAccountId ?? defaultAccountId)!,
+        maybeAccountId!,
```

Comment:
Since we are inside a `useEffect` that early exits if `maybeAccountId` is undefined, we can kill the not null assertion here.

Reply:
Trivial, so I fixed it in https://github.com/shapeshift/web/pull/3724/commits/fa96c635f6251b31386b1a3b109b3de18d3ec56e.